nuxeo-runtime/nuxeo-runtime-mongodb/src/main/java/org/nuxeo/runtime/mongodb/MongoDBComponent.java

```diff
@@ -103,8 +110,107 @@ public MongoDatabase getDatabase(String id) {
     public Iterable<MongoDatabase> getDatabases() {
         return () -> clients.entrySet().stream().map(e -> {
             MongoDBConnectionConfig c = getDescriptor(XP_CONNECTION, e.getKey());
-            return MongoDBConnectionHelper.getDatabase(e.getValue(), c.dbname);
+            return recordDatabaseVersion(e.getValue(), c);
         }).iterator();
     }
 
+    /**
+     * Records the version of the mongo server containing the database and returns the database.
+     *
+     * @param client the mongo client
+     * @param config the mongo connection config descriptor
+     * @return the mongo database after recording the mongo server version.
+     * @since 10.10-HF16
+     */
+    protected MongoDatabase recordDatabaseVersion(MongoClient client, MongoDBConnectionConfig config) {
```

Comment:
I guess this could be just called `getDatabase` then.

Reply:
But it doesn't matter, this is just a 10.10-HF code and is not in master.

Reply:
There is already a `getDatabase` and `getDatabases` which both call this `recordDatabaseVersion`.
I propose to call it `internalGetDatabase` like in some other codes

Reply:
The existing `getDatabase` and `getDatabases` methods have different signatures. I don't see what's bad about having a `protected` overload with a different signature.

Reply:
As I said it doesn't really matter given that this code is not in master

Reply:
Sold !